### PR TITLE
Add evalml-core

### PR DESCRIPTION
**Pull request by freddyaboulton**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/evalml-core/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    - shap >=0.35.0
+    - texttable >=1.6.2
+    - python >=3.6.*
+
+test:
+  imports:
+    - evalml
+  requires:
+    - pytest ==4.4.1
+    - nbval ==0.9.3
```
Comment (scopatz) [inline recipes/evalml-core/meta.yaml:41]: Do these need to be so tight?

Comment (freddyaboulton) [reply recipes/evalml-core/meta.yaml:43]: I relaxed pytest! I'm not sure about nbval but I don't want to stray from what's listed in the package's test requirements.

Comment (scopatz) [reply recipes/evalml-core/meta.yaml:43]: OK

File: recipes/evalml-core/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  dev_url: https://github.com/Featuretools/featuretools/
+  home: https://www.alteryx.com/innovation-labs
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE
+  summary: EvalML is an AutoML library that builds, optimizes, and evaluates machine learning pipelines using domain-specific objective functions.
+
+extra:
+  recipe-maintainers:
+    - dsherry
```
Comment (scopatz) [inline recipes/evalml-core/meta.yaml:64]: @dsherry are you OK with being a maintainer?

Comment (dsherry) [reply recipes/evalml-core/meta.yaml:64]: @scopatz yes! Thank you.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**